spec/workflow/payload_template_spec.rb

```diff
@@ -20,6 +20,16 @@
       end
     end
 
+    context "with intrinsic functions" do
+      let(:payload) { {"uuid.$" => "States.UUID()"} }
+      let(:context) { {} }
+      let(:inputs)  { {} }
+
+      it "returns the value from the inputs" do
```

Comment:
this statement is slightly off - I think a better test for "value from the inputs" is to use maybe `States.Array($.input)` and give it actual input.  Otherwise this is more like "executes the intrinsic function".

Reply:
Oh yeah my bad copy & paste from below, fixed